tools/clang/lib/Sema/SemaCXXScopeSpec.cpp

```diff
@@ -200,9 +201,13 @@ bool Sema::RequireCompleteDeclContext(CXXScopeSpec &SS,
 
   // If we're currently defining this type, then lookup into the
   // type is okay: don't complain that it isn't complete yet.
+  // HLSL: If this is a builtin type, add all method definitions and complete
+  // the type.
   QualType type = Context.getTypeDeclType(tag);
   const TagType *tagType = type->getAs<TagType>();
-  if (tagType && tagType->isBeingDefined())
+  if (tagType && tagType->isBeingDefined() &&
+      (hlsl::IsUserDefinedRecordType(type) ||
+       hlsl::IsHLSLObjectWithImplicitMemberAccess(type)))
```

Comment:
This change doesn't make sense to me. Can you explain why this is required?

If we get here and `tagType` a builtin type that is not yet completed, we should go down into `RequireCompleteType` below which will call into the external sema source to complete it. I'm confused why we need additional early-outs here, and the comment above makes it more confusing.

Also changes in this file should be wrapped in `// HLSL Change Begin|End` comments.

Reply:
The current HLSL-specific modifications to scoped lookup do not support 'static'. I've changed my approach and eagerly complete the HitObject type (including its static functions) instead. This means, no special handling for HitObject is any longer necessary.
This specific change is gone now - considering this resolved.